sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -503,7 +503,9 @@ class Dataset[T] private[sql](
    * @group basic
    * @since 1.6.0
    */
-  def schema: StructType = queryExecution.analyzed.schema
+  def schema: StructType = sparkSession.withActive {
+    queryExecution.analyzed.schema
```

Comment:
does this operation involve active session?

Reply:
Unfortunately it does. We were seeing errors in one of the thrift suites, the behavior of some expressions depends on SQLConf.get. In this particular case `IntegralDivide` is the offender. This is actually super bad since it can lead to correctness problems, and we should fix this in more a complete way.